drivers/adc/adc_stm32.c

```diff
@@ -1174,6 +1176,34 @@ static const struct adc_driver_api api_stm32_driver_api = {
 	.ref_internal = STM32_ADC_VREF_MV, /* VREF is usually connected to VDD */
 };
 
+/* st_prescaler property requires 2 elements : clock ASYNC/SYNC and DIV */
+#define STM32_ADC_CLOCK(x, prop) DT_INST_PROP_BY_IDX(x, prop, 0)
+#define STM32_ADC_DIV(x, prop) DT_INST_PROP_BY_IDX(x, prop, 1)
+
+/* macro to check is 1st element is SYNC or ASYNC */
+#define STM32_ADC_CLOCK_IS_SYNC(x, prop) IS_EQ(STM32_ADC_CLOCK(x, prop), SYNC)
+
+/* Macro to set the prefix depending on the 1st element */
+#define STM32_ADC_DO_PREFIX(x, prop)				\
+	COND_CODE_1(IS_EQ(STM32_ADC_CLOCK(x, prop), SYNC),	\
+		prescaler_sync,					\
+		prescaler_async)
```

Comment:
```suggestion
#define STM32_ADC_PREFIX(x, prop)				\
	COND_CODE_1(IS_EQ(STM32_ADC_CLOCK(x, prop), SYNC),	\
		LL_ADC_CLOCK_SYNC_PCLK_DIV,					\
		LL_ADC_CLOCK_ASYNC_DIV)
```

Reply:
Using directly LL_ADC_xxx prefix to simplify and change macro name